sql/catalog/lease: TestAsOfSystemTimeUsesCache failed

sql/catalog/lease.TestAsOfSystemTimeUsesCache [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17019167?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17019167?buildTab=artifacts#/) on release-24.2 @ [7668752918959c9e1ad645ff6fd57dc05bae13bc](https://github.com/cockroachdb/cockroach/commits/7668752918959c9e1ad645ff6fd57dc05bae13bc):


```
=== RUN   TestAsOfSystemTimeUsesCache
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestAsOfSystemTimeUsesCache1700005910
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x235732c]

goroutine 63093 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x4005c776a8?, {0x63a9958, 0x40129d36d0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:231 +0x68
panic({0x4676f40?, 0x93525f0?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorState).upsertLeaseLocked(0x400a29baa0, {0x63a9958, 0x40129d36d0}, {0x64410f8, 0x4005606000}, {0x40016e3ef0?, 0x16e3f08?}, {0x63a2870, 0x400ea87340}, {0x9446cd0, ...})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/descriptor_state.go:188 +0x39c
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x63a9958, 0x40129d36d0})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/lease.go:686 +0x2c8
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x63a9958?, 0x40129d36d0?})
	github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:389 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x4004d393b0, {0x63a9958, 0x40129d36d0}, {0x6346e78?, 0x4004141508?}, 0x4002605f40)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:320 +0xac
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x4001452060, {0x63a9990?, 0x4006295260?}, 0x400f8e4b00, {0x40125f2f10, 0x3}, {0x4004d393b0?, 0x58?}, 0x400e5ae580)
	github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:388 +0x244
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 62466
	github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:358 +0x484
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAsOfSystemTimeUsesCache.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42488

--- a/pkg/sql/catalog/lease/descriptor_state.go
+++ b/pkg/sql/catalog/lease/descriptor_state.go
@@ -185,7 +185,9 @@ func (t *descriptorState) upsertLeaseLocked(
 	// is subsumed we have nothing to delete. In dual-write mode clearing
 	// this guarantees only the old expiry based lease is cleaned up. In
 	// Session only clearing this means the release is a no-op.
-	toRelease.sessionID = nil
+	if toRelease != nil {
+		toRelease.sessionID = nil
+	}
 	return nil, toRelease, nil
 }
 

